src/coreclr/jit/codegenlinear.cpp

```diff
@@ -2133,8 +2133,8 @@ void CodeGen::genProduceReg(GenTree* tree)
                     }
                 }
             }
-#if FEATURE_ARG_SPLIT
-            else if (tree->OperIsPutArgSplit())
+#if FEATURE_ARG_SPLIT_SUPPORTED
+            else if (compFeatureArgSplit() && tree->OperIsPutArgSplit())
```

Comment:
Presumably, in cases like this we don't need to check for `compFeatureArgSplit() && ` because we would never have inserted a GT_PUTARG_SPLIT tree node unless it was legal?